### PR TITLE
api: remove ssl

### LINKED ISSUE #301: Disable SSL by default
Disable SSL support by default (replace `go_tarantool_ssl_disable` tag with `go_tarantool_ssl_enable`).
SSL is used in less than 50% cases, but it breaks cross-compilation due to the use of cgo.

### PR DESCRIPTION
**Pull request by DerekBum**
`OpenSslDialer` and all of its helper functions and tests were rellocated to the `go-tlsdialer` [1] package (and renamed to `OpenSSLDialer`). So now we can safely remove all the copy-pasted code from `go-tarantool`.

This way, in order to use SSL, user should import the `go-tlsdialer` package and call functions from there.

After all Tarantool-EE specific logic was moved to the `go-tlsdialer`, Tarantool-EE jobs could be removed from ci workflows. In a separate commit, Tarantool-EE jobs are removed from the ci workflows.

1. https://github.com/tarantool/go-tlsdialer/

Related issues:

Closes #301

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -169,6 +165,57 @@ otherwise it will have a description which can be retrieved with `err.Error()`.
 by the method `Do` of object `conn` which is the object that was returned
 by `Connect()`.
 
+### Example with openssl
+
+To use the SSL dialer a
```
Comment (oleg-jukovec) [inline README.md:170]:
```suggestion
To use the OpenSSL dialer a
```

File: README.md [hunk 2]
```diff
@@ -169,6 +165,57 @@ otherwise it will have a description which can be retrieved with `err.Error()`.
 by the method `Do` of object `conn` which is the object that was returned
 by `Connect()`.
 
+### Example with openssl
```
Comment (oleg-jukovec) [inline README.md:168]:
```suggestion
### Example with encrypting traffic
```
https://www.tarantool.io/en/doc/latest/enterprise/security/#encrypting-traffic

File: README.md [hunk 3]
```diff
@@ -169,6 +165,57 @@ otherwise it will have a description which can be retrieved with `err.Error()`.
...
+	).Get()
+	if err != nil {
+		fmt.Printf("Error: %s", err)
+	} else {
+		fmt.Printf("Data: %v", data)
+	}
+}
+```
+
+Note that SSL is enabled only on Tarantool Enterprise Edition 2.10 or newer.
```
Comment (oleg-jukovec) [inline README.md:217]:
```suggestion
Note that [traffic encryption](https://www.tarantool.io/en/doc/latest/enterprise/security/#encrypting-traffic)
is enabled only on Tarantool Enterprise Edition 2.10 or newer.
```

File: README.md [hunk 4]
```diff
@@ -315,8 +362,10 @@ and user may cancel it in process.
 
 Now you need to pass `Dialer` as the second argument instead of URI.
 If you were using a non-SSL connection, you need to create `NetDialer`.
-For SSL-enabled connections, use `OpenSslDialer`. Please note that the options
-for creating a connection are now stored in corresponding `Dialer`, not in `Opts`.
+For SSL-enabled connections, use `tlsdialer.OpenSslDialer` of the `go-tlsdialer`
```
Comment (oleg-jukovec) [inline README.md:365]:
```suggestion
For SSL-enabled connections, use `OpenSslDialer` from the `go-tlsdialer`
```

File: dial_test.go [hunk 5]
```diff
@@ -485,9 +469,7 @@ func testDialAccept(opts testDialOpts, l net.Listener) chan dialServerActual {
 
 		// Read Auth request.
 		authRequestExpected := authRequestExpectedChapSha1
-		if opts.isPapSha256Auth {
-			authRequestExpected = authRequestExpectedPapSha256
```
Comment (oleg-jukovec) [inline dial_test.go:489]: Please, add:

* Unit tests with `PapSha256Auth` for `AuthDialer`.
* Integration test with Tarantool CE and NetDialer + Greeting + Protocol + Auth with `PapSha256Auth` (Tarantool should return an error).

**Comment by oleg-jukovec** [comment]
In the commit message:

s/[1] https://github.com/tarantool/go-tlsdialer/1. https://github.com/tarantool/go-tlsdialer/

The commit message should be < 72 characters. Now it is too short at the first glance.

File: dial_test.go (hunk 5 above)
Comment (DerekBum) [reply dial_test.go:489]: `NetDialer` does not have the `Auth` field, so user cannot choose the auth method for this dialer. It is set to always be `ChapSha1Auth`.

Comment (oleg-jukovec) [reply dial_test.go:489]: I think we could try to re-auth:

dialer := AuthDialer{
    Dialer: NetDialer{
    },
}

Comment (DerekBum) [reply dial_test.go:489]: New tests added.

File: README.md (hunk 4 above)
Comment (DerekBum) [reply README.md]: Updated.

File: README.md (hunk 3 above)
Comment (DerekBum) [reply README.md]: Updated.

File: README.md (hunk 2 above)
Comment (DerekBum) [reply README.md]: Updated.

File: README.md (hunk 1 above)
Comment (DerekBum) [reply README.md]: Updated.

File: dial_test.go [hunk 6]
```diff
@@ -754,6 +749,39 @@ func TestAuthDialer_Dial(t *testing.T) {
...
+		Auth:     tarantool.PapSha256Auth,
+	}
+	ctx, cancel := test_helpers.GetConnectContext()
+	defer cancel()
+	conn, err := authDialer.Dial(ctx, tarantool.DialOpts{})
+	if conn != nil {
+		conn.Close()
+	}
+
+	assert.Nil(t, err)
```
Comment (oleg-jukovec) [inline dial_test.go:780]:
```suggestion
	assert.NoError(t, err)
```

File: tarantool_test.go [hunk 7]
```diff
@@ -790,6 +790,27 @@ func TestNetDialer_BadUser(t *testing.T) {
...
+	ctx, cancel := test_helpers.GetConnectContext()
+	defer cancel()
+	conn, err := authDialer.Dial(ctx, DialOpts{})
+	if conn != nil {
+		conn.Close()
+		t.Fatalf("Connection created successfully")
+	}
+
+	assert.NotNil(t, err)
+	assert.ErrorContains(t, err, "failed to authenticate")
```
Comment (oleg-jukovec) [inline tarantool_test.go:811]: It should be enough.
```suggestion
	assert.ErrorContains(t, err, "failed to authenticate")
```

**Review by oleg-jukovec — APPROVED** [review]
Thank you for the patch!

File: dial_test.go (hunk 6 above)
Comment (DerekBum) [reply dial_test.go]: Updated.

File: tarantool_test.go (hunk 7 above)
Comment (DerekBum) [reply tarantool_test.go]: Updated.

File: README.md [hunk 8]
```diff
@@ -169,6 +165,58 @@ otherwise it will have a description which can be retrieved with `err.Error()`.
...
+	if err != nil {
+		fmt.Printf("Error: %s", err)
+	} else {
+		fmt.Printf("Data: %v", data)
+	}
+}
+```
+
+Note that [traffic encryption](https://www.tarantool.io/en/doc/latest/enterprise/security/#encrypting-traffic)
+is enabled only on Tarantool Enterprise Edition 2.10 or newer.
```
[the remark this answers is not in the document]
Comment (DerekBum) [reply README.md]: Updated.

File: CHANGELOG.md [hunk 9]
```diff
@@ -15,7 +15,6 @@ Versioning](http://semver.org/spec/v2.0.0.html) except to the first release.
 - IsNullable flag for Field (#302)
 - More linters on CI (#310)
 - Meaningful description for read/write socket errors (#129)
-- Support password and password file to decrypt private SSL key file (#319)
```
Comment (DifferentialOrange) [inline CHANGELOG.md:18]: Wait, shouldn't there also be "move SSL support to a separate module"?

**Comment by oleg-jukovec** [comment]
Please, `s/OpenSsl/OpenSSL/g`

File: CHANGELOG.md (hunk 9 above)
Comment (DerekBum) [reply CHANGELOG.md:18]: Added the `CHANGELOG` entry.

File: README.md [hunk 10]
```diff
@@ -169,6 +165,58 @@ otherwise it will have a description which can be retrieved with `err.Error()`.
 by the method `Do` of object `conn` which is the object that was returned
 by `Connect()`.
 
+### Example with encrypting traffic
+
+To use the OpenSSL dialer a
```
[the remark this answers is not in the document]
Comment (DerekBum) [reply README.md]: Updated.

File: CHANGELOG.md [hunk 11]
```diff
@@ -125,6 +124,7 @@ Versioning](http://semver.org/spec/v2.0.0.html) except to the first release.
 - Code() method from the Request interface (#158)
 - `Schema` field from the `Connection` struct (#7)
 - `OkCode` and `PushCode` constants (#237)
+- Move SSL support to a separate module `go-tlsdialer` (#301)
```
Comment (DifferentialOrange) [inline CHANGELOG.md:127]: It was simply
> ### Removed
> - IPROTO constants

for moving IPROTO constants to a separate package. I think here we too should simply
> ### Removed
> - SSL support (#301)

**Review by DifferentialOrange — APPROVED** [review]

File: CHANGELOG.md (hunk 11 above)
Comment (DerekBum) [reply CHANGELOG.md]: Ok, updated.

**Pull request merged**